docs/blog/contributor-development-maintenance-guide.en-US.md

```diff
@@ -0,0 +1,155 @@
+
+   ```bash
+   # Run the test command to update snapshot
+   npm run test -u
+   ```
+
+2. However, if the changes during snapshot comparison exceed the scope of your modification this time, for example, you only changed the text, but found that classnames in snapshots have changed, which obviously does not meet expectations, we need to rectify the causes.
+
+   The common causes are as follows:
+
+   - Local dependency is too old, it is possible to pull the latest code, but did not update the local dependency, resulting in the dependency package version is too low and the output results are inconsistent.The
+
+     solution is as simple as deleting the lock file、node_modules，and reinstall dependencies.
+
+   - You may have locally changed the core code in addition to the demo, causing the logic to change. You need to go over this change carefully.
+
+### rc-x Library dependencies
+
+In Ant Design, most components are an upper encapsulation of a component based on `react-component`. Therefore, if a user reports a bug, when we troubleshoot the problem, if we find the problem is in the `@rc-component/xxx` or `rc-xxx` component, Then we need to put pr on those components to fix it.After the fix, we need to verify the fix results in the Ant Design project, then we can link the project to the Ant Design project for verification.For example:
+
+Run `npm link` under the rc project.
+
+![image](https://user-images.githubusercontent.com/10286961/224603053-98488c2d-f33c-4c25-8c09-6c790cfcdbf6.png)
+
+Run `npm link "Project name"` in Ant Design
+
+![image](https://user-images.githubusercontent.com/10286961/224603065-95715727-83d0-4ef9-81e4-3b7065aaf73e.png)
+
+Once we have verified, we can mention pr to the rc component.It should be noted that link may cause an exception when running the test command. Therefore, after local verification, we need to run the following
+
+command locally to delete the package from link:
+
+```bash
+npm unlink "rc-field-form" --no-save
+npm i
+```
+
+When the pr is finally merged, usually the maintainer will release a version. If the patch version is released, then you only need to install and verify it in the Ant Design project, but if the minor version is released, Then we also need to take the initiative to upgrade the version in the Ant Design project, and after passing the local verification, a separate pr to Ant Design for upgrade repair.
+
+### Property expiration
+
+In a large project, if you want to scrap an property or method, it is actually very troublesome, because your project may already have a lot of items in use, if you kill it, then others upgrade the console screen will be red, or even can't run.But in the process of project iteration, we will encounter more and more scenes and problems, a long time ago solution may no longer fit, really need to scrap, then, we need to adopt a soft, less radical way to scrap, leave enough time for users to modify.
```

Comment:
```suggestion
In a large project, if you want to remove an property or a method, it is actually very troublesome. Since your project may already have a lot of items in use, other projects depend on it may get broken. But as the project iterated, we will encounter more and more problems which solutions long time ago may no longer fit. Then, we need to adopt a soft, less radical way to remove it, leaving enough time for users to modify.
```